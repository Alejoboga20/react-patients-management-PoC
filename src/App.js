import React, { useState, useEffect } from 'react';
import { Appoinment } from './components/Appoinment';
import { Form } from './components/Form';

export const App = () => {
  let initialAppoinments = JSON.parse(localStorage.getItem('appoinments'));

  if (!initialAppoinments) {
    initialAppoinments = [];
  }

  const [appoinmentArray, setAppoinmentArray] = useState(initialAppoinments);

  useEffect(() => {
    if (initialAppoinments) {
      localStorage.setItem('appoinments', JSON.stringify(appoinmentArray));
    } else {
      localStorage.setItem('appoinments', JSON.stringify([]));
    }
  }, [appoinmentArray, initialAppoinments]);

  const handleAddAppoinment = (appoinment) =>
    setAppoinmentArray([...appoinmentArray, appoinment]);

  const handleDeleteById = (id) => {
    const filteredAppoinments = appoinmentArray.filter(
      (appoinment) => appoinment.id !== id
    );

    setAppoinmentArray(filteredAppoinments);
  };

  return (
    <div>
      <h1>Patients Manager</h1>

      <div className='container'>
        <div className='row'>
          <div className='one-half column'>
            <Form handleAddAppoinment={handleAddAppoinment} />
          </div>
          <div className='one-half column'>
            <h2>
              {appoinmentArray.length
                ? 'Manage your Appoinments'
                : 'Add an Appoinment'}
            </h2>

            {appoinmentArray.map((appoinment) => (
              <Appoinment
                appoinment={appoinment}
                key={appoinment.id}
                handleDeleteById={handleDeleteById}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};
