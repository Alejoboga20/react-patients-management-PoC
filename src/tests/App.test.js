import { mount } from 'enzyme';
import { App } from '../App';

const handleAddAppoinment = jest.fn();

describe('App Tests', () => {
  const wrapper = mount(<App />);

  beforeEach(() => jest.clearAllMocks());

  test('should match snaphot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have correct title when no appoinments are added', () => {
    expect(wrapper.find('h2').at(1).text()).toBe('Add an Appoinment');
  });

  test('should not call handleAddAppoinment with no inputs', () => {
    wrapper.find('button').simulate('click');

    expect(handleAddAppoinment).toHaveBeenCalledTimes(0);
  });
});
