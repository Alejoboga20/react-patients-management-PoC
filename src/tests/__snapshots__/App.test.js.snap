// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests should match snaphot 1`] = `
<div>
  <h1>
    Patients Manager
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="one-half column"
      >
        Array [
          <h2>
            Create Appoinment
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <label>
              Pet's Name
            </label>
            <input
              className="u-full-width"
              name="pet"
              onChange={[Function]}
              placeholder="Pet's Name"
              type="text"
              value=""
            />
            <label>
              Owner's Name
            </label>
            <input
              className="u-full-width"
              name="owner"
              onChange={[Function]}
              placeholder="Owner's Name"
              type="text"
              value=""
            />
            <label>
              Date
            </label>
            <input
              className="u-full-width"
              name="date"
              onChange={[Function]}
              type="date"
              value=""
            />
            <label>
              Time
            </label>
            <input
              className="u-full-width"
              name="time"
              onChange={[Function]}
              type="time"
              value=""
            />
            <label>
              Symptoms
            </label>
            <textarea
              className="u-full-width"
              name="symptoms"
              onChange={[Function]}
              value=""
            />
            <button
              className="u-full-width button-primary"
              type="submit"
            >
              Add Appoinment
            </button>
          </form>,
        ]
      </div>
      <div
        className="one-half column"
      >
        <h2>
          Add an Appoinment
        </h2>
      </div>
    </div>
  </div>
</div>
`;
